docs/proposals/11-cluster-health.md

```diff
@@ -0,0 +1,205 @@
+
+If there is a way to detect the exact reason for this condition (for example, the loadbalancer is down), then that information could be published as a separate condition type.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty if this condition lasts for some duration.
+- The `kube-controller-manager` could be scaled down to avoid marking `nodes` and `pods` as `NotReady` unnecessarily.
+This could be done by the scaler part of the current `probe` [sub-command](https://github.com/gardener/dependency-watchdog/blob/master/cmd/probe.go) of the [dependency-watchdog](https://github.com/gardener/dependency-watchdog) but re-factored as a separate scaler component which scales some targets based on some configured conditions in the `Cluster` resource. 
+
+#### ETCDUnreachable
+
+The `Endpoint` monitoring part of the `root` [command](https://github.com/gardener/dependency-watchdog/blob/master/cmd/root.go) of the [dependency-watchdog](https://github.com/gardener/dependency-watchdog) can be re-factored into a separate component that updates the conditions with the `ETCDUnreachable` condition type.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty if this condition lasts for some duration.
+- When this condition transitions from status `true` to `false` (i.e. `etcd` transitions from being unreachable to reachable), the dependent `pods` could be deleted if they are in `CrashLoopBackoff`. This is to make sure the dependent `pods` recover as fast as possible from an `etcd` down time. This could be done by the restarter part of the current `root` [command](https://github.com/gardener/dependency-watchdog/blob/master/cmd/root.go) of the [dependency-watchdog](https://github.com/gardener/dependency-watchdog) but re-factored as a separate restarter component which restarts (deletes) some targets `pods` based on some configured conditions in the `Cluster` resource but only if the `pods` are in `CrashLoopBackoff`. 
+
+#### KubeAPIServerUnreachableInternally
+
+The `KubeAPIServerUnreachableInternally` condition type can be implemented (both the monitoring and the remedial action parts) similar to the [`ETCDUnreachable`](#etcdunreachable) condition type.
+
+#### UnhealthyNodesThresholdReached
+
+A component can monitor the fraction of the `NotReady` nodes in a zone and update the conditions with the `UnhealthyNodesThresholdReached` condition type.
+The list of affected zones could be maintained in the [`Message`](https://github.com/kubernetes/cluster-registry/blob/master/pkg/apis/clusterregistry/v1alpha1/types.go#L168).
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty if this condition lasts for some duration.
+- If necessary, `kube-controller-manager` could be scaled down as described [above](#possible-actions).
```

Comment:
Do not do this, you are risking to break down the healthy part of the cluster.

Reply:
> Do not do this, you are risking to break down the healthy part of the cluster.

Agreed. Hence, the *If necessary* caveat. I presume that `kube-controller-manager` would act sanely and not go berserk and mark workloads `NodeReady` willy-nilly if sizeable number of nodes go unhealthy. But some parts of the `nodelifecycle` controller don't inspire confidence.